unified-codes/apps/data-service/src/app/AuthenticationService.ts

```diff
@@ -0,0 +1,24 @@
+import { JWTToken } from './JWT';
+import IdentityProvider from './IdentityProvider';
+import User from './User';
+
+export interface IUserCredentials {
+  username: string;
+  password: string;
+}
+
+export class AuthenticationService {
+  provider: IdentityProvider;
+
+  constructor(provider: IdentityProvider) {
+    this.provider = provider;
+  }
+
+  // TODO: add headers or whatever we need to pass the credentials?
```

Comment:
Oops, this `// TODO` is wrong (that stuff should be managed by the identity provider!)